tour/main.js

```diff
@@ -85,9 +101,14 @@ const fetchUserSettings = async (SCRIPT_API) => {
     method: 'GET',
   });
   const data = await response.json();
+  cachedUserSettings = data;
   return data;
 };
 
+export function updateCachedUserData(newUserData) {
+  cachedUserSettings = { ...cachedUserSettings, ...newUserData };
```

Comment:
Ideally there's a function that updates the userSettings with new data (POST request) and also does this so you don't have to call `updateCachedUserData` every time you make a userSettings POST request.